src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -7152,24 +7154,425 @@ public class MyAttribute : System.Attribute
+
+            cPosition = getIdentifierPosition("c");
+            Assert.Equal("System.String", parentModel.GetSpeculativeTypeInfo(cPosition, newNameOf, SpeculativeBindingOption.BindAsExpression).Type.ToTestDisplayString());
+
+            dPosition = getIdentifierPosition("d");
+            Assert.Equal("TParameter", parentModel.GetSpeculativeTypeInfo(dPosition, newNameOfArgument, SpeculativeBindingOption.BindAsExpression).Type.ToTestDisplayString());
+
+            return;
+
+            int getIdentifierPosition(string identifier)
+            {
+                return tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(i => i.Identifier.ValueText == identifier).Single().SpanStart;
+            }
+
+            static ExpressionSyntax parseNameof(string source, CSharpParseOptions parseOptions)
+                => SyntaxFactory.ParseCompilationUnit($@"{source};", options: parseOptions).DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+
+            static ExpressionSyntax parseIdentifier(string source, CSharpParseOptions parseOptions)
+                => SyntaxFactory.ParseCompilationUnit($@"{source};", options: parseOptions).DescendantNodes().OfType<IdentifierNameSyntax>().Single();
+        }
+
+        [Fact]
+        public void TypeParameterScope_InMethodAttributeNameOf_SpeculatingWithReplacementAttribute()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        local<object>();
+
+        [My(a)]
+        void local<TParameter>() { }
+    }
+
+    [My(b)]
+    void M2<TParameter>() { }
+}
+
+public class MyAttribute : System.Attribute
+{
+    public MyAttribute(string name1) { }
+}
+";
+            // C# 10
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
+                // (8,13): error CS0103: The name 'a' does not exist in the current context
+                //         [My(a)]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(8, 13),
+                // (12,9): error CS0103: The name 'b' does not exist in the current context
+                //     [My(b)]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "b").WithArguments("b").WithLocation(12, 9)
+                );
+
+            var tree = comp.SyntaxTrees.Single();
+            var parentModel = comp.GetSemanticModel(tree);
+            var localFuncPosition = tree.GetText().ToString().IndexOf("[My(a)]", StringComparison.Ordinal);
+            var methodPosition = tree.GetText().ToString().IndexOf("[My(b)]", StringComparison.Ordinal);
+
+            var attr = parseAttributeSyntax("[My(nameof(TParameter))]", TestOptions.Regular10);
+            VerifyTParameterSpeculation(parentModel, localFuncPosition, attr, found: false);
```

Comment:
>VerifyTParameterSpeculation(parentModel, localFuncPosition, attr, found: false); [](http://example.com/codeflow?start=12&length=80)

I do not think this tests scenario 2B because the location is supposed to be **inside** an existing attribute. The location used here is not inside the existing attribute

Reply:
I'd wrote down the following:
> Scenario 2B:
> TryGetSpeculativeSemanticModel with a new attribute

The reason for this is that the only `TryGetSpeculativeSemanticModel` overloads take statements, attributes, type syntaxes, etc, but I didn't find one that just takes a plain expressions.

Reply:
>The reason for this is that the only TryGetSpeculativeSemanticModel overloads take statements, attributes, type syntaxes, etc, but I didn't find one that just takes a plain expressions.


I am not following. I am talking about position where to speculate, not about with what syntax to speculate.

Reply:
I see. We can try that, although it seems strange to speculate with an attribute inside an attribute.

Reply:
I don't know. One might arrive to that by thinking: "I want to see what happens if I replace existing attribute application with a new one." And passes location related to the attribute about to be replaced.

Reply:
Added a test